src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -155,7 +155,8 @@ pub fn compile_ws<'a>(
         return Compilation::new(&bcx);
     }
     if options.build_config.dry_run {
-        return Compilation::new(&bcx);
+        let build_runner = BuildRunner::new(&bcx)?;
+        return build_runner.dry_run();
```

Comment:
I'm seeing other `build_runner.compile` checks that might be worth having, like collisions.  Should we be like `unit_graph` and be in the `build_context` and have `build_runner.compile` decide how much dry-run behavior to do?

Reply:
I added the `collision` check.
Maybe I'm not understanding this correctly but according to me we can't do like `unit_graph` because we need to call method that are private to the `BuildRunner` class and are needed notably for getting the destination path of the executable we are installing.